apps/os/backend/trpc/routers/integrations.ts

```diff
@@ -1164,4 +1164,595 @@ export const integrationsRouter = router({
+
+      // 7. Create trial channel and send invite
+      // Note: User sync happens just-in-time when external users send messages
+      const { createTrialSlackConnectChannel } = await import("../../utils/trial-channel-setup.ts");
+      const result = await createTrialSlackConnectChannel({
+        db: ctx.db,
+        userEstateId: estate.id,
+        userEmail,
+        userName,
+        iterateTeamId,
+        iterateBotToken: iterateBotAccount.accessToken,
+      });
+
+      if (!result.success) {
+        throw new Error(`Something went wrong while setting up trial`, { cause: result.error });
+      }
+
+      logger.info(
+        `Successfully set up trial for ${userEmail}: channel ${result.channelName} → estate ${estate.id}`,
+      );
+
+      return {
+        success: true,
+        estateId: estate.id,
+        organizationId: organization.id,
+        channelId: result.channelId,
+        channelName: result.channelName,
+      };
+    }),
+
+  /**
+   * Upgrades a trial estate to a full Slack installation
+   * This removes all trial-specific configuration so the user can connect their own Slack workspace
+   */
+  upgradeTrialToFullInstallation: protectedProcedure
+    .input(
+      z.object({
+        estateId: z.string(),
+      }),
+    )
+    .mutation(async ({ ctx, input }) => {
+      const { estateId } = input;
+
+      logger.info(`Upgrading trial estate ${estateId} to full installation`);
+
+      // Verify this is actually a trial estate
+      const estate = await ctx.db.query.estate.findFirst({
+        where: eq(schemas.estate.id, estateId),
+        columns: {
+          organizationId: true,
+        },
+      });
+
+      if (!estate) {
+        throw new TRPCError({
+          code: "NOT_FOUND",
+          message: "Estate not found",
+        });
+      }
+
+      const { getSlackChannelOverrideId } = await import("../../utils/trial-channel-setup.ts");
+      const trialChannelId = await getSlackChannelOverrideId(ctx.db, estateId);
+      if (!trialChannelId) {
+        throw new TRPCError({
+          code: "BAD_REQUEST",
+          message: "This estate is not a trial estate",
+        });
+      }
+
+      // Verify user has permission to modify this estate
+      const membership = await ctx.db.query.organizationUserMembership.findFirst({
+        where: and(
+          eq(schemas.organizationUserMembership.organizationId, estate.organizationId),
+          eq(schemas.organizationUserMembership.userId, ctx.user.id),
+        ),
+      });
+
+      if (!membership) {
+        throw new TRPCError({
+          code: "FORBIDDEN",
+          message: "You don't have permission to modify this estate",
+        });
+      }
+
+      const iterateTeamId = ctx.env.SLACK_ITERATE_TEAM_ID;
+      if (!iterateTeamId) {
+        throw new TRPCError({
+          code: "INTERNAL_SERVER_ERROR",
+          message: "Iterate Slack workspace not configured",
+        });
```

Comment:
no need to throw `INTERNAL_SERVER_ERROR`, we can just `throw new Error("Iterate Slack workspace not configured")`, trpc maps these anyway